pkg/server/field_mask.go

```diff
@@ -22,3 +24,84 @@ func parseFieldMask(message proto.Message, paths []string) (model.Mask, error) {
 
 	return parsedMask.Paths, nil
 }
+
+// updateFromMask updates the dst with the values from src based on the mask.
+//
+// Following scenarios are possible:
+//   - mask is nil: All fields are updated.
+//   - mask is not nil, but has no paths: No fields are updated.
+//   - mask contains paths, that does not exist in the model: The paths are ignored.
+//   - mask contains paths, that exist in model: Only those fields are updated.
+//
+// `protoName` tags are used to match fields from the fieldMask to fields in the model.
+func updateFromMask[T any](src, dst *T, mask model.Mask) *T {
+	// If mask is nil, update all fields
+	if mask == nil {
+		return src
+	}
+
+	for _, path := range mask {
+		fields := strings.Split(path, ".")
+		updateField(reflect.ValueOf(src).Elem(), reflect.ValueOf(dst).Elem(), fields)
+	}
+
+	return dst
+}
+
+// updateField updates the dst value with the values from src, based on the fields slice.
+func updateField(src, dst reflect.Value, fields []string) {
+	if len(fields) == 0 {
```

Comment:
What happens if fields contain wildcard character '*' ?

Reply:
if i remember correctly 
"google.golang.org/protobuf/types/known/fieldmaskpb"

does not support `*`